packages/pigeon/lib/src/kotlin/kotlin_generator.dart

```diff
@@ -475,7 +475,9 @@ class KotlinGenerator extends StructuredGenerator<InternalKotlinOptions> {
 
     void writeEncodeLogic(EnumeratedType customType) {
       final String encodeString =
-          customType.type == CustomTypes.customClass ? 'toList()' : 'raw';
+          customType.type == CustomTypes.customClass
```

Comment:
@tarrinneal At some point we should consider converting statements like these to the new expression switches (`final String encodeString = switch (customType.type) {...};`), so that if we ever add another custom type the compiler will find all the places we need to adjust the logic, instead of having the new type randomly opted into one or the other existing codepath everywhere.